Source/WebCore/css/parser/CSSPropertyParserConsumer+Align.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2016-2023 Apple Inc. All rights reserved.
+ * Copyright (C) 2024 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CSSPropertyParserConsumer+Align.h"
+
+#include "CSSContentDistributionValue.h"
+#include "CSSParserIdioms.h"
+#include "CSSParserTokenRange.h"
+#include "CSSPropertyParserConsumer+Ident.h"
+#include "CSSValueKeywords.h"
+#include "CSSValuePair.h"
+#include <optional>
+
+namespace WebCore {
+namespace CSSPropertyParserHelpers {
+
+using IsPositionKeyword = bool (*)(CSSValueID);
```

Comment:
You moved this and did not create it. But note that this type sounds a little bit like a boolean or an enumeration we use as a named boolean and programmers could make the mistake of using it as a boolean, which would compare against null so turn into `true` and probably not even warn. So maybe tweak the name slightly?

Reply:
Renamed to `PositionKeywordPredicate`.